clippy_lints/src/utils/internal_lints.rs

```diff
@@ -207,3 +218,29 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for LintCollector<'a, 'tcx> {
         NestedVisitorMap::All(&self.cx.tcx.hir)
     }
 }
+
+
+
+pub struct DefaultHashTypes;
+
+impl LintPass for DefaultHashTypes {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(PREFER_FX_HASH)
+    }
+}
+
+impl EarlyLintPass for DefaultHashTypes {
+    fn check_ident(&mut self, cx: &EarlyContext<'_>, ident: Ident) {
+        let msg = {
+            let ident: String = ident.to_string();
+            if ident.contains("HashMap") && !ident.contains("FxHashMap") {
```

Comment:
Another solution to consider for this would be to have a `HashMap` with the keys being the disallowed `ident`s (strings) and the value the wanted replacement.

Then you can just 
```rust
let msg = if let Some(replace) = map.get(ident_string) {
    format!("Prefer {} over {}", replace, ident_string);
} else {
    return
}
```

This would also be easy to extend, if needed.

Reply:
What's the `to_string` of an `Ident` that `.contains("HashMap")`? Is it `"HashMap"`? Or is it something more fully-qualified?

Reply:
It should just be `"HashMap" `.